t/33-developer_mode.t

```diff
@@ -258,8 +258,9 @@ subtest 'developer session visible in live view' => sub {
 
 subtest 'status-only route accessible for other users' => sub {
     $driver->get('/logout');
+    $driver->element_text_is('#user-action a', 'Login', 'logged-out before logging in as otherdeveloper');
     $driver->get('/login?user=otherdeveloper');
-    is $driver->find_element('#user-action a')->get_text(), 'Logged in as otherdeveloper', 'otherdeveloper logged-in';
+    $driver->element_text_is('#user-action a', 'Logged in as otherdeveloper', 'otherdeveloper logged-in');
```

Comment:
This doesn't change anything. Internally it will just be a sequence of `find_element` + `get_text`. I've just looked at the code:

```
sub _find_element_with_action {
…
    my $element;
    eval { $element = $self->find_element( $locator, $locator_strategy ); };
    if ($@) {
        print "# Error: $@\n";
        return 0;
    }
    return $element->$method( $params, $desc );
}

sub element_text_is {
    my $self   = shift;
    my $method = 'text_is';
    return $self->_find_element_with_action( $method, @_ );
}
```

If your repeated runs show a difference then the sample size is likely too small.

Note that this function also does not introduce a loop or anything in order to wait until the text matches the expected value. Maybe that's what we needed to do - although I'm not quite sure yet.

Reply:
The actual fix is line 261 which explicitly checks that the expected login is there. This is what I tried to explain in the git commit message https://github.com/os-autoinst/openQA/pull/5137/commits/dce1b8e510cd6e7faa9f8615b256becd3eeebee7 . The actual effect might just be the same as a small "sleep" call making it more likely to show the login button after time. I am sure that this actually only reduces the likelyhood back to values we had some time ago. The test is still unstable.